Keas.Mvc/ClientApp/components/Workstations/AssignWorkstation.tsx

```diff
@@ -24,258 +24,212 @@ interface IProps {
+  const confirmClose = () => {
     if (!confirm('Please confirm you want to close!')) {
       return;
     }
 
-    this._closeModal();
+    closeModal();
   };
 
-  private _closeModal = () => {
-    this.setState({
-      date: addYears(startOfDay(new Date()), 3),
-      error: {
-        message: '',
-        path: ''
-      },
-      person: null,
-      submitting: false,
-      validState: false,
-      workstation: null
+  const closeModal = () => {
+    setDate(addYears(startOfDay(new Date()), 3));
+    setError({
+      message: '',
+      path: ''
     });
-    this.props.closeModal();
+    setPerson(null);
+    setWorkstation(null);
+    setSubmitting(false);
+    setValidState(false);
+    props.closeModal();
   };
 
   // assign the selected workstation even if we have to create it
-  private _assignSelected = async () => {
-    if (!this.state.validState || this.state.submitting) {
+  const assignSelected = async () => {
+    if (!validState || submitting) {
       return;
     }
 
-    this.setState({ submitting: true });
-    const person = this.props.person ? this.props.person : this.state.person;
-    const workstation = this.state.workstation;
+    setSubmitting(true);
+    const personData = props.person ? props.person : person;
 
     try {
-      await this.props.onCreate(
-        person,
-        workstation,
-        format(this.state.date, 'MM/dd/yyyy')
-      );
+      await props.onCreate(personData, workstation, format(date, 'MM/dd/yyyy'));
     } catch (err) {
-      this.setState({ submitting: false });
+      setSubmitting(false);
       return;
     }
-    this._closeModal();
+    closeModal();
   };
 
   // once we have either selected or created the workstation we care about
-  private _onSelected = (workstation: IWorkstation) => {
-    this.setState({ workstation }, this._validateState);
+  const onSelected = (workstation: IWorkstation) => {
+    setWorkstation(workstation);
   };
 
-  private _onDeselected = () => {
-    this.setState(
-      {
-        error: {
-          message: '',
-          path: ''
-        },
-        workstation: null
-      },
-      this._validateState
-    );
+  const onDeselected = () => {
+    setError({
+      message: '',
+      path: ''
+    });
+    setWorkstation(null);
   };
 
-  private _onSelectPerson = (person: IPerson) => {
-    this.setState({ person }, this._validateState);
+  const onSelectPerson = (person: IPerson) => {
+    setPerson(person);
   };
 
-  private _validateState = () => {
-    const error = yupAssetValidation(
-      workstationSchema,
-      this.state.workstation,
-      {}, // no context
-      { date: this.state.date, person: this.state.person }
-    );
-    this.setState({ error, validState: error.message === '' });
+  const changeDate = (newDate: Date) => {
+    setDate(newDate);
   };
 
-  private _changeDate = (newDate: Date) => {
-    this.setState({ date: startOfDay(new Date(newDate)) }, this._validateState);
```

Comment:
was the startOfDay method not doing anything or just not important?

Reply:
Wait I forgot to add the start of day method when changing the date